client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx

```diff
@@ -89,24 +89,21 @@ const getTitanMenuItems = ( {
 			} ),
 			onClick: getTitanClickHandler( 'webmail' ),
 		},
-		...( canCurrentUserAddEmail( domain )
-			? [
-					{
-						isInternalLink: true,
-						materialIcon: 'delete',
-						onClick: () => {
-							showRemoveMailboxDialog?.();
-
-							recordTracksEvent( 'calypso_email_management_titan_remove_mailbox_click', {
-								domain_name: mailbox.domain,
-								mailbox: mailbox.mailbox,
-							} );
-						},
-						key: `remove_mailbox:${ mailbox.mailbox }`,
-						title: translate( 'Remove mailbox' ),
-					},
-			  ]
-			: [] ),
+		{
+			isInternalLink: true,
+			materialIcon: 'delete',
+			disabled: ! canCurrentUserAddEmail( domain ),
+			onClick: () => {
+				showRemoveMailboxDialog?.();
+
+				recordTracksEvent( 'calypso_email_management_titan_remove_mailbox_click', {
+					domain_name: mailbox.domain,
+					mailbox: mailbox.mailbox,
+				} );
+			},
+			key: `remove_mailbox:${ mailbox.mailbox }`,
+			title: translate( 'Remove mailbox' ),
+		},
```

Comment:
This change makes it so that we always display the `Remove mailbox` item, but disable it when the current user doesn't have permission to delete the mailbox.

This was something I experimented with while trying out different solutions for the action menu and I found it makes sense. Given that we already display a warning on the email management page when users aren't allowed to add new mailboxes, I think it should be fairly clear why it's disabled (see screenshot in PR description to see what it looks like for users without permission).